File: gatormicrophone.ts
/**
* Andy England @ SparkFun Electronics
* September 6, 2018

* Development environment specifics:
* Written in Microsoft Makecode
* Tested with a SparkFun gatorMicrophone sensor and micro:bit
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/


/**
 * Functions to operate the gator:microphone sensor
 */
//% color=#f44242 icon="\uf130"
namespace gatorMicrophone {
    // Functions for reading envelope and gate data
	
	/**
	* Reads the intensity of sound from the microphone as a value between 0 and 2047, sensitivity is determined by the switch on the gator:microphone
	*/
	//% weight=25
	//% blockId="gatorMicrophone_getSoundIntensity"
	//% block="sound intensity"
	//% shim=gatorMicrophone::getSoundIntensity
	export function getSoundIntensity(): number{
		return 0
	}
	
	/**
	* Returns true if the sound is above a certain level, sensitivity is determined by the switch on the gator:microphone
	*/
	//% weight=24
	//% blockId="gatorMicrophone_readGateData"
	//% block="check for sound"
	//% shim=gatorMicrophone::readGateData
	export function readGateData(): boolean{
		return true
	}
	
	/**
	* Sets the gain, or volume of the microphone. Lower gain means more sensitivity to smaller changes in volume.
	*/
	//% weight=23
	//% blockId="gatorMicrophone_setGain"
	//% block="change gain to %GainOptions"
	//% shim=gatorMicrophone::setGain
	//% advanced=true
	export function setGain(gainOption: GainOptions){
		return
	}
}

File: gatormicrophone.cpp
/**
* Andy England @ SparkFun Electronics
* September 6, 2018
* https://github.com/sparkfun/pxt-light-bit
*
* Development environment specifics:
* Written in Microsoft PXT
* Tested with a SparkFun temt6000 sensor and micro:bit
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/


#include "pxt.h"
#include <cstdint>
#include <math.h>
#include "SparkFun_ADS1015_Arduino_Library.h"

enum GainOptions
{
	//% block="Two Thirds" enumval=0
	Two_Thirds = 0,
	//% block="One" enumval=1
	One = 1,
	//% block="Two" enumval=2
	Two = 2,
	//% block="Four" enumval=3
	Four = 3,
	//% block="Eight" enumval=4
	Eight = 4,
	//% block="Sixteen" enumval=5
	Sixteen = 5,
};

using namespace pxt;

namespace gatorMicrophone {
	ADS1015 *mic; //Create an ADS1015 sensor	

	//%
	uint16_t getSoundIntensity()
	{
		uint16_t envelopeData = mic->getSingleEnded(2);
		//release_fiber();
		return envelopeData;
	}
	
	//%
	bool readGateData()
	{
		uint16_t gateData = mic->getSingleEnded(3);
		if (gateData > 1000)
		{
			//release_fiber();
			return true;
		}
		return false;
	}
	
	//%
	void setGain(uint16_t gainOption)
	{
		mic->setGain(gainOption);
	}
}